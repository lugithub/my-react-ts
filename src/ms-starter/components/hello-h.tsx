import React, { useState } from "react";

import { HelloProps, getExclamationMarks } from "./hello";

export function Helloh({ name }: HelloProps) {
  const [{ enthusiasmLevel }, setState] = useState({
    enthusiasmLevel: 1
  });

  function handleClick() {
    setState(prevS => {
      return {
        enthusiasmLevel: prevS.enthusiasmLevel + 1
      };
    });
  }
  return (
    <div className="hello" onClick={handleClick}>
      <div className="greeting">
        Hello {name + getExclamationMarks(enthusiasmLevel)}
      </div>
    </div>
  );
}
