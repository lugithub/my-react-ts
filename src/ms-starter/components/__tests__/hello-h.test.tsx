import React from "react";
import { shallow, mount } from "enzyme";
import { Helloh } from "../hello-h";

it("renders the correct text when no enthusiasm level is given", () => {
  const hello = shallow(<Helloh name="Daniel" />);
  expect(hello.find(".greeting").text()).toEqual("Hello Daniel!");
});

it("renders the correct text when no enthusiasm level is given, snapshot", () => {
  const hello = shallow(<Helloh name="Daniel" />);
  expect(hello).toMatchSnapshot();
});

it("renders the correct text with an explicit enthusiasm level of 2", () => {
  const hello = mount(<Helloh name="Daniel" />);
  hello.find(".hello").simulate("click");
  expect(hello.find(".greeting").text()).toEqual("Hello Daniel!!");
});
