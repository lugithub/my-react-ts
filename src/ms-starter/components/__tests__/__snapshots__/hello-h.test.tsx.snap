// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct text when no enthusiasm level is given, snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Helloh
    name="Daniel"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="greeting"
      >
        Hello 
        Daniel!
      </div>,
      "className": "hello",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Hello ",
          "Daniel!",
        ],
        "className": "greeting",
      },
      "ref": null,
      "rendered": Array [
        "Hello ",
        "Daniel!",
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="greeting"
        >
          Hello 
          Daniel!
        </div>,
        "className": "hello",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello ",
            "Daniel!",
          ],
          "className": "greeting",
        },
        "ref": null,
        "rendered": Array [
          "Hello ",
          "Daniel!",
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
